ingest/indexer/service/indexer_streaming_service.go

```diff
@@ -120,54 +122,55 @@ func (s *indexerStreamingService) publishTxn(ctx context.Context, req abci.Reque
 		fee := feeTx.GetFee()
 
 		// Message type
-		// TO BE VERIFIED - This may not be the correct way to obtain message type
 		txMessages := tx.GetMsgs()
-		msgType := txMessages[0].String()
-
-		// Include these events only:
-		// - token_swapped
-		// - pool_joined
-		// - pool_exited
-		// - create_position
-		// - withdraw_position
-		events := res.GetEvents()
-		var includedEvents []domain.EventWrapper
-		for i, event := range events {
-			clonedEvent := deepCloneEvent(&event)
-			// Add the token liquidity to the event
-			err := s.addTokenLiquidity(ctx, clonedEvent)
-			if err != nil {
-				s.logger.Error("Error adding token liquidity to event", "error", err)
-				return err
+		msgType := proto.MessageName(txMessages[0])
+		txResults := res.GetTxResults()
+		for _, txResult := range txResults {
+			events := txResult.GetEvents()
+			// Iterate through the events in the transaction
+			// Include these events only:
+			// - token_swapped
+			// - pool_joined
+			// - pool_exited
+			// - create_position
+			// - withdraw_position
+			var includedEvents []domain.EventWrapper
+			for i, event := range events {
+				clonedEvent := deepCloneEvent(&event)
+				// Add the token liquidity to the event
+				err := s.addTokenLiquidity(ctx, clonedEvent)
+				if err != nil {
+					s.logger.Error("Error adding token liquidity to event", "error", err)
+					return err
+				}
+				err = s.adjustTokenInAmountBySpreadFactor(ctx, clonedEvent)
+				if err != nil {
+					s.logger.Error("Error adjusting amount by spread factor", "error", err)
+					continue
+				}
+				eventType := clonedEvent.Type
+				if eventType == gammtypes.TypeEvtTokenSwapped || eventType == gammtypes.TypeEvtPoolJoined || eventType == gammtypes.TypeEvtPoolExited || eventType == concentratedliquiditytypes.TypeEvtCreatePosition || eventType == concentratedliquiditytypes.TypeEvtWithdrawPosition {
```

Comment:
nit: this could be encapsulated by a separate function to improve readability

Reply:
Good point. Will address it as fly-by in other further PR to avoid too many backward / forward porting exercises.